src/scry/completion/method_db.cr

```diff
@@ -53,39 +57,54 @@ module Scry::Completion
 
   class Visitor < Crystal::Visitor
     property classes
+    property class_name = ""
```

Comment:
I think it's better to use `nil` to denote the absence of `class_name`, so the compiler can help you instead of doing empty checks.

Reply:
I think it's easier with an empty string cause we only interpolate stuff, otherwise I'll have to use not_nil! calls

Reply:
@class_name is assigned  the value of the generate_classname which will be an empty string if the queues are empty.

Reply:
```crystal
property class_name : String?
```

and

```crystal
if this_class_name = class_name
  # Use this_cass_name here...
end
```

WDYT?

Reply:
Although I see no issues with initializing the `class_name = ""` 

@bew Is this a bad practice or something? :sweat_smile:

Reply:
I'll simply say:

```
method_receiver = node.receiver ? "#{@class_name}.class" : @class_name
```
In this case, does it make sense (allowed) to have `@class_name` as an empty string?
I don't think so, so it must be made difficult/impossible to do.

Reply:
that won't happen cause if class_name is "" then class_queue is empty and in here we opt out https://github.com/crystal-lang-tools/scry/pull/100/files/d05348c07392320c379be63575985d0f83398c56#diff-45d877a34e1040cf2af4c4c46cdcf290R73
The idea of the classname is just to cache the output of generate_class_name which never returns nil, so initialising it as nil doesn't seem like the right way to handle it.

Reply:
@laginha87 Oh, ok, I think `class_name = ""` is not that bad @bew WDYT?

Reply:
yeah let's not block the PR for a tiny non-straight-forward thing like that, it's ok